### PR TITLE
Support Flex Message update 2

### LINKED ISSUE #299: Support Flex Message Update 2
News: https://developers.line.biz/en/news/2020/10/08/flex-message-update-2-released/

1. - [ ] [Changed the maximum number of bubbles that can be included in a carousel](https://developers.line.biz/en/reference/messaging-api/#f-carousel)
2. - [ ] [Box component has justifyContent and alignItems property](https://developers.line.biz/en/reference/messaging-api/#box)
3. - [ ] [Use linear gradient as the background image for a box component by setting its background.type property equal to linearGradient.](https://developers.line.biz/en/reference/messaging-api/#box)
4. - [ ] [Allowed an empty array to be specified as the value of a box's contents property](https://developers.line.biz/en/reference/messaging-api/#box)
5. - [ ] [Added support for animated images](https://developers.line.biz/en/reference/messaging-api/#f-image)
6. - [ ] [Changed how sizes are specified](https://developers.line.biz/en/news/2020/10/08/flex-message-update-2-released/#update-size)
	1. - [ ] [Image component](https://developers.line.biz/en/docs/messaging-api/flex-message-layout/#image-size)
	2. - [ ] [Icon, text, and span components](https://developers.line.biz/en/docs/messaging-api/flex-message-layout/#other-component-size) 
7. - [ ] [Allowed text to automatically shrink to fit a component's width](https://developers.line.biz/en/news/2020/10/08/flex-message-update-2-released/#update-adjust-mode) - [Text](https://developers.line.biz/en/reference/messaging-api/#f-text) and [Button](https://developers.line.biz/en/reference/messaging-api/#button)
8. - [ ] [Spacer has been discontinued](https://developers.line.biz/en/news/2020/10/08/flex-message-update-2-released/#update-spacer)

### PR DESCRIPTION
**Pull request by louis70109**
fixed #299 

- Box component: justifyContent and alignItems property
- Background & linearGradient model
- Button `shrink-to-fit` property
- Remove SpacerComponent

### CODE REVIEW

File: linebot/models/background.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        """
+        super(Background, self).__init__(**kwargs)
+
+        self.type = None
+
+
+class LinearGradientBackground(Background):
+    """LinearGradientBackground."""
+
+    def __init__(self, angle=None, start_color=None, end_color=None,
```
Comment (cryeo) [inline linebot/models/background.py:42]: [must]
`angle`, `start_color` and `end_color` are required when background.type is linearGradient.

File: linebot/models/flex_message.py [hunk 2]
```diff
@@ -548,6 +567,8 @@ class SpacerComponent(FlexComponent):
 
     This is an invisible component that places a fixed-size space
     at the beginning or end of the box
+
+    SpacerComponent is deprecated.
```
Comment (cryeo) [inline linebot/models/flex_message.py:571]: [ask]
Should we keep `SpacerComponent` class? (because its status is discontinued, not deprecated)

Comment (cryeo) [reply linebot/models/flex_message.py:571]: [fyi]
Some codes related to `SpacerComponent` are still remaining.

https://github.com/line/line-bot-sdk-python/blob/5ab6ba225495248c52eb3aa728da427bedffb37d/tests/models/serialize_test_case.py#L44
https://github.com/line/line-bot-sdk-python/blob/2ff8fe138ef616b0b46696881a6e09afd6871b69/linebot/models/flex_message.py#L323

**Review by cryeo — CHANGES_REQUESTED** [review]

File: linebot/models/flex_message.py (hunk 2 above)
Comment (louis70109) [reply linebot/models/flex_message.py:571]: Ok, I think we need to remove `SpacerComponent` class.

**Comment by louis70109** [comment]
Thanks for @cryeo -san details review!!
I have fixed these.

**Review by cryeo — APPROVED** [review]

**Pull request merged**